homeassistant/components/mqtt/mixins.py

```diff
@@ -207,6 +209,16 @@ def validate_device_has_at_least_one_identifier(value: ConfigType) -> ConfigType
     )
 
 
+def validate_sensor_entity_category(config: ConfigType) -> ConfigType:
+    """Check the sensor's entity category is `diagnostic` or `None`."""
+    if (
+        CONF_ENTITY_CATEGORY in config
+        and config[CONF_ENTITY_CATEGORY] == EntityCategory.CONFIG
```

Comment:
This check does not match the docstring.
To make this future proof, maybe it's better to implement like the docstring suggests?

Reply:
Updated the docstr as the check is making sure the entitity category cannot be `config`.